File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        left, c_idx, right = 0, 0, len(nums) - 1

        while c_idx <= right:
            c_num = nums[c_idx]
            if c_num == 2:
                nums[c_idx], nums[right] = nums[right], nums[c_idx]
                right -= 1
            
            elif c_num == 0:
                nums[c_idx], nums[left] = nums[left], nums[c_idx]

                left += 1
                c_idx += 1
            else:
                c_idx += 1

File: 0075-sort-colors/0075-sort-colors.cs
public class Solution {
    public void SortColors(int[] nums) {
        int l = 0;
        int r = nums.Length - 1;
        int m = 0;
        while (m <= r) {
            Print(nums);
            if (nums[m] == 1) {
                m++;
            } else if (nums[m] == 0) {
                int tmp = nums[l];
                nums[l] = nums[m];
                nums[m] = tmp;
                l++;
                m++;
            } else {
                int tmp = nums[r];
                nums[r] = nums[m];
                nums[m] = tmp;
                r--;
            }
        }       
    }
    public void Print(int[] nums) {
        foreach(var num in nums) Console.Write(num + " ");
        Console.WriteLine();
    }
}